Exporter Update and rewrite

pcarranza opened the pull request:
In case you want to take it, I basically did the following: 

- rewrite the whole thing using a newer version of everything
- add a release process to have versions
- add more metrics for allocations, node versions, resources, etc
- send a pr to nomad API to make it fail also on read timeout
- document more
- remove vendor in favor of managing dependencies with dep.
- add an apache2 license.

pcarranza committed:
Merge deployment metrics in, it's rough and raw, but ¯\_(ツ)_/¯

pcarranza committed:
Fix compilation issues

pcarranza committed:
Cleaning up, setting units, rinse and repeat

pcarranza committed:
Now this makes more sense

pcarranza committed:
Add goreleaser basics

pcarranza committed:
Add docker build instructions

pcarranza committed:
Changed the unit of some metrics

pcarranza committed:
Update goreleaser manifest

pcarranza committed:
Move version to specific package

pcarranza committed:
Fix dependencies, fix leader detection, fix some metrics units

pcarranza committed:
Remove vendor folder, use dep

pcarranza committed:
Update README, and changed a couple things on the help texts

pcarranza committed:
Adjust the default timeout, ensure that we can read node stats.

pcarranza committed:
Only read metrics from leader, or it is allowed to read stale metrics

pcarranza committed:
Update the readme with the latest information and arguments

pcarranza committed:
Add a couple more metrics.

pcarranza committed:
Add nomad API supported version

pcarranza committed:
Add apache2 License

pcarranza committed:
Move port to the reserved one

pcarranza committed:
Use ConstMetric in all the places that we can, set up and leader always

pcarranza committed:
Rename metric so it's prometheus style compliant

pcarranza committed:
Remove _total endings from guages

pcarranza committed:
Merge pull request #1 from pcarranza/pc/refactor-metrics-names

Address Brian's feedback

pcarranza committed:
Remove allocations and stats to reduce the number of api queries

pcarranza committed:
Add options to disable any collection

pcarranza committed:
Allow stale consistency for everything except the leader query

pcarranza committed:
Enable stats metrics again but allow guarding behind a flag

pcarranza committed:
Ignore version less than 0.8.1 when pulling allocations metrics

pcarranza committed:
Add the min version logic to allocations

pcarranza committed:
Set the wait time in the configuration so it gets sent to the server

pcarranza committed:
Fix concurrency problem when skipping nodes due to version

pcarranza committed:
Fix concurrency problems, for real this time

pcarranza committed:
Reshape some metrics so they make more sense

pcarranza committed:
Revert "Merge pull request #1 from pcarranza/pc/refactor-metrics-names"

This reverts commit 3a3bf3d6df7beba10cca14f4484f2f00726ad496, reversing
changes made to bb5b761274dde981c28d593d894f049acabc6100.

pcarranza committed:
Merge branch 'master' into pc/spike-to-identify-server-dance

pcarranza committed:
Merge pull request #2 from pcarranza/pc/spike-to-identify-server-dance

Fix concurrency problems

pcarranza committed:
Merge branch 'master' of github.com:pcarranza/nomad-exporter

pcarranza committed:
Skip nodes completely when it's down

pcarranza committed:
Add a bunch of latency metrics, invert node search

pcarranza committed:
Add a description so we can in fact describe, duh

pcarranza committed:
Debug how much time we are using just in case it's necessary

pcarranza committed:
Set wait time to all the client requests

pcarranza committed:
Only query for nodes once

pcarranza committed:
Add pprof for further troubleshooting

pcarranza committed:
Change summaries to histograms

pcarranza committed:
Change timeouts and split wait timeout from read timeout

pcarranza committed:
if err is not nil, then node is nil, duh

pcarranza committed:
Remove dead code

pcarranza committed:
Merge pull request #4 from pcarranza/pc/skip-nodes-completely-when-down

Skip nodes completely when down

Nomon commented:
Hi, I have recently had no time to give to any of my github projects and I do not see that changing anytime soon. Would you be willing to take over the project if I transfer the ownership to you? For now I have added you to the collaborators, feel free to merge what you think is necessary.

pcarranza commented:
@Nomon cool, thanks, will do.

pcarranza committed:
Add version to all the alloc labels

pcarranza committed:
Fix task labels order, there's a nasty connascence of position

pcarranza committed:
Merge pull request #6 from pcarranza/pc/add-version-to-alloc-labels

Add version to alloc labels

Commit:
Add a counter for zombie allocs and skip them

Commit:
Remove labels from allocation_zombies

Commit:
Add some Serf metrics

pcarranza committed:
Merge pull request #9 from omame/omame/serf_metrics

Add some Serf metrics

pcarranza committed:
Revert "Fix task labels order, there's a nasty connascence of position"

This reverts commit a016f73ce0a861e57e420257ad28903cbf297851.

pcarranza committed:
Revert "Add version to all the alloc labels"

This reverts commit d3231b5efcc953598e34b71a1a93975beac7301c.

pcarranza committed:
Merge pull request #8 from omame/omame/zombie-allocs

Add a counter for zombie allocs and skip them

dgonzalezruiz committed:
Add extra label to metric so similar hostnames don't collide

pcarranza committed:
Merge pull request #10 from dgonzalezruiz/fix-clonic-hostname

Add extra label to metric so similar hostnames don't collide

Commit:
Fix the latest tagging due to deprecation in goreleaser

pcarranza committed:
Merge pull request #11 from pcarranza/omame/fix-goreleaser

Fix goreleaser syntax

eunomie committed:
Fix nodeUsedMemory metric value

The `nodeStats.Memory.Used` value, provided by nomad API is already in bytes.
So it's not needed to multiply it by `1024*1024` to have a good value.

With this change the `nodeUsedMemory` uses the same unit than
`nodeAllocatedMemory` and `nodeResourceMemory` and can be compared to.

Tested against a nomad 0.8.7.
Not easy to find the real type in nomad source code. The memory value is computed using https://github.com/shirou/gopsutil/blob/master/mem

pcarranza committed:
Merge pull request #13 from squarescale/node-used-memory-bytes

Fix nodeUsedMemory metric value

dgonzalezruiz committed:
Skip allocations which desired state is not equal to run

This should cause nomad-exporter to stop exporting metrics from allocations in `terminal` state.

Fetching and exporting metrics related to these allocations makes `nomad-exporter` start using high amounts of resources, and it has little purpose

pcarranza committed:
Merge pull request #14 from dgonzalezruiz/patch-1

Skip allocations which desired state is not equal to run

dgonzalezruiz committed:
Remove desired_status label

This label is now useless, as it will be fixed to `run` after the change in https://github.com/pcarranza/nomad-exporter/pull/14

pcarranza committed:
Merge pull request #15 from dgonzalezruiz/patch-2

Remove desired_status label

dgonzalezruiz committed:
Actually remove desired_status label from nomad exporter metrics

After https://github.com/pcarranza/nomad-exporter/pull/15 was merged, nomad-exporter crashes upon start due to the metrics being built with less labels than the ones expected here. This fixes it.

pcarranza committed:
Merge pull request #17 from dgonzalezruiz/patch-3

Actually remove desired_status label from nomad exporter metrics

pcarranza committed:
Update nomad api version

pcarranza committed:
Update docker base image version

pcarranza committed:
Add gitlab-ci file

pcarranza committed:
Update makefile, and package path

pcarranza committed:
Extract args outside and configure the http client

pcarranza committed:
Fix error message

pcarranza committed:
Rename file to make more sense

pcarranza committed:
Add job version to the deployments, allocations and tasks

pcarranza committed:
Merge branch 'pc/update-nomad-api' into 'master'

Update nomad api

See merge request yakshaving.art/nomad-exporter!1

pcarranza committed:
Expose port in the dockerfile

pcarranza committed:
Expose port in the dockerfile

pcarranza committed:
Add nomad runner file

pcarranza committed:
Merge branch 'pc/add-nomad-runner-file' into 'master'

Add nomad runner file

See merge request yakshaving.art/nomad-exporter!2

pcarranza committed:
Fix format of the hcl file

pcarranza committed:
Add job version to nomad allocation, update README

pcarranza committed:
Update README file to the latest data

pcarranza committed:
Remove failed boolean that is basically useless

pcarranza committed:
Update goreleaser configuration

pcarranza committed:
Merge branch 'pc/update-nomad-api' into 'master'

Expose port in the dockerfile

See merge request yakshaving.art/nomad-exporter!3

pcarranza committed:
Add status endpoint

pcarranza committed:
Merge branch 'pc/add-status-endpoint' into 'master'

Add status endpoint

See merge request yakshaving.art/nomad-exporter!4

pcarranza committed:
OCD compliant TLS variable name

obourdon committed:
Add support for Nomad standard environment variables

The environment variables are used as default if set
and can be superseded by command line arguments if needed.

pcarranza committed:
Merge branch 'env-variables' into 'master'

Add support for Nomad standard environment variables

See merge request yakshaving.art/nomad-exporter!5

pcarranza committed:
Add documentation & change the vars so they are more golike

obourdon committed:
Fix dumb conditional error

pcarranza committed:
Merge branch 'fix-dumb-cond-error' into 'master'

Fix dumb conditional error

See merge request yakshaving.art/nomad-exporter!6

Commit:
Merge branch 'patch-1' into 'master'

Add node_id label to node_info metric in order to avoid metric collissions

See merge request yakshaving.art/nomad-exporter!7

ciberRoomer committed:
Add node_id label to node_info metric in order to avoid metric collissions

ciberRoomer committed:
Fix exported labels after node_info changes

pcarranza committed:
Merge branch 'patch-1' into 'master'

Fix exported labels after node_info changes

See merge request yakshaving.art/nomad-exporter!8

pcarranza committed:
Update Dockerfile

draoncc committed:
Fix missing httpClient TLS configuration

pcarranza committed:
Merge branch 'master' into 'master'

Fix missing httpClient TLS configuration

See merge request yakshaving.art/nomad-exporter!10
